mips_api/__init__.py

```diff
@@ -1,11 +1,152 @@
-from mips_api import mips
-
 import json
 import os
 
+import boto3
+import requests
+
+
+_mips_url_login = 'https://login.abilaonline.com/api/v1/sso/mipadv/login'
+_mips_url_chart = 'https://mipapi.abilaonline.com/api/v1/maintain/chartofaccounts'
+_mips_url_logout = 'https://mipapi.abilaonline.com/api/security/logout'
+
+# This is global so that it can be stubbed in test.
+# Because this is global its value will be retained
+# in the lambda environment and re-used on warm runs.
+ssm_client = None
+
+
+def _get_os_var(varnam):
+    try:
+        return os.environ[varnam]
+    except KeyError as exc:
+        raise Exception(f"The environment variable '{varnam}' must be set")
+
+
+def _parse_omit_codes(omit_codes):
+    return omit_codes.split(',')
+
+
+def _parse_extra_codes(extra_codes):
+    data = {}
+    for _kv_pair in extra_codes.split(','):
+        k, v = _kv_pair.split(':', 1)
+        data[k] = v
+    return data
+
+def collect_secrets(ssm_path):
+    '''Collect secure parameters from SSM'''
+
+    # create boto client
+    global ssm_client
```

Comment:
we need to use the global keyword in order to modify the variable, otherwise we would create a shadow copy on line 42 instead of changing the global value: https://www.w3schools.com/python/python_variables_global.asp

Reply:
thanks for the info.  amazingly this is news to me.